prboom2/src/SDL/i_system.c

```diff
@@ -254,11 +254,27 @@ const char *I_DoomExeDir(void)
 
 const char* I_GetTempDir(void)
 {
-  static char tmp_path[PATH_MAX] = {0};
+  static const char* tmp_path = NULL;
 
-  if (tmp_path[0] == 0)
+  if (!tmp_path)
   {
-    GetTempPath(sizeof(tmp_path), tmp_path);
+    wchar_t wpath[PATH_MAX];
+    DWORD result;
+
+#ifdef HAVE_GET_TEMP_PATH_2
+    result = GetTempPath2W(PATH_MAX, wpath);
+#else
+    result = GetTempPathW(PATH_MAX, wpath);
+#endif
+    if (result == 0 || result > MAX_PATH)
+    {
+      lprintf(LO_ERROR, "GetTempPath(2) failed\n");
+      tmp_path = strdup(".");
```

Comment:
If the current directory is used as a temporary directory, unzipping a file in the same directory may fail, e.g.:

- `dsda-doom -iwad doom2 -file ./foo.zip`
- `GetTempPathW` fails and `I_GetTempDir` returns `.`
- `D_AddZip` uses `./foo.zip/` as the output directory:
```c
dsda_InitString(&temporary_directory, I_GetTempDir());
dsda_StringCatF(&temporary_directory, "/%s/", dsda_BaseName(zipped_file_name));
M_MakeDir(temporary_directory.string, true);
```

Maybe strip the `.zip` part of the name with `dsda_CutExtension`.

Reply:
We can't use `.` ever, this would be completely insane if files just get dumped into a directory without being requested by a user.

Reply:
Could you leave this branch as an `I_Error` for this PR? I will set up something in the dsda_doom_data directory as a fallback, but raising an error for now is fine with me.

Reply:
Done in https://github.com/kraflab/dsda-doom/pull/304/commits/f4c81bc197c81bc7072445122ef5ce8dd27d7780